### PR TITLE
Matching

### PR DESCRIPTION
**Pull request by BrandonWade**
Adds additional fields to the registration page needed for user matching

### CODE REVIEW

**Comment by paymog** [comment]
Need to write tests.

**Comment by paymog** [comment]
I specifically didn't test the change to UserController because we're moving away from it with `LoginLanding`

**Comment by kyleleung75** [comment]
Everything looked good through the files, I am still going to run it locally and make sure everything is good.

**Comment by paymog** [comment]
Sounds good

**Comment by kyleleung75** [comment]
AHH the matching is beautiful! It resets the user too when you reclick the matching button. The tests were good too. I suppose we should think about making a timeout for it so that it can send back a message for them to try again later, but everything looks great!

**Comment by paymog** [comment]
Sick. The only problem is refreshing the page at any point after you've
started the matching process. I tried to come up with a good solution but
failed. However, I think I've though of one since I put up the PR. I'll
implement that once the landing page goes through.

**Comment by kjrcda** [comment]
Okay cool. I should have it done tomorrow. I may have to redo my changes on a new branch. rebasing now is a ****\* because I've done it 3 times already. I'll evaluate tomorrow which is quicker (rebate or new branch) and get it done ASAP. Shouldn't take more than 1hr either way.

-------- Original message --------

File: webapp/ng_tests/controllers/home_matching_controller_test.js [hunk 1]
```diff
@@ -42,17 +42,27 @@ describe('UserMatchingController', function () {
...
+        });
+
+        it('should update the user cookie on hasBeenMatched', function(){
+            createController();
+
+            var userResponse = {user: { email: CURRENT_USER}};
+            socket.receive('hasBeenMatched', userResponse);
+
+            expect(authService.setUser).toHaveBeenCalledWith(userResponse.user)
         })
```
Comment (kjrcda) [inline webapp/ng_tests/controllers/home_matching_controller_test.js:66]: ;

File: webapp/ng_tests/controllers/home_matched_controller_test.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        authService.currentUser.and.returnValue({matchedWith: FIRST_NAME});
+    }));
+
+    describe('on initialization', function() {
+        it('sets $scope.name based on authService', function() {
+            createController();
+
+            expect(authService.currentUser).toHaveBeenCalled();
+            expect($rootScope.name).toBe(FIRST_NAME);
+        })
```
Comment (kjrcda) [inline webapp/ng_tests/controllers/home_matched_controller_test.js]: ;

File: webapp/ng_tests/controllers/home_page_controller_test.js [hunk 3]
```diff
@@ -23,27 +23,42 @@ describe('HomePageController', function () {
...
+        });
+
+        it('navigates to matching when matching is set in cookie', function() {
+            var authService = jasmine.createSpyObj('authService', ['currentUser']);
+            authService.currentUser.and.returnValue({wantsToBeMatched: true});
+
+            $state.expectTransitionTo('home.matching');
+
+            createController(authService);
         })
```
Comment (kjrcda) [inline webapp/ng_tests/controllers/home_page_controller_test.js]: ;

File: webapp/ng_tests/controllers/main_page_controller_test.js [hunk 4]
```diff
@@ -90,7 +94,7 @@ describe('MainPageController', function () {
             //note do not expectPOST
 
             expect($rootScope.errorMessages).toBeFalsy();
-            expect(authService.login.calls.any()).toBeFalsy();
+            expect(authService.setUser.calls.any()).toBeFalsy();
         })
```
Comment (kjrcda) [inline webapp/ng_tests/controllers/main_page_controller_test.js]: ;

File: webapp/ng_tests/controllers/main_page_controller_test.js [hunk 5]
```diff
@@ -100,7 +104,7 @@ describe('MainPageController', function () {
             $state.expectTransitionTo('register');
 
             $rootScope.createAccount();
-            expect(authService.login.calls.any()).toBeFalsy();
+            expect(authService.setUser.calls.any()).toBeFalsy();
         })
```
Comment (kjrcda) [inline webapp/ng_tests/controllers/main_page_controller_test.js]: ; and line below

File: webapp/ng_tests/controllers/register_controller_test.js [hunk 6]
```diff
@@ -129,7 +165,7 @@ describe('RegisterController', function () {
             $httpBackend.flush();
 
             expect($rootScope.errorMessages).toBe(errorMessage);
-            expect(authService.login.calls.any()).toBeFalsy();
+            expect(authService.setUser.calls.any()).toBeFalsy();
         })
```
Comment (kjrcda) [inline webapp/ng_tests/controllers/register_controller_test.js]: ;

File: webapp/public/javascripts/controllers.js [hunk 7]
```diff
@@ -67,31 +72,44 @@ lunchrControllers.controller('RegisterController', ['$scope', '$http', '$state',
...
+        socket.on('hasBeenMatched', function(data){
+            var userObject = angular.fromJson(data.user);
+            authService.setUser(userObject);
+        });
+
+
+        socket.on('matched' + currentUser.email, function (data) {
+            var userObject = angular.fromJson(data.user);
+            authService.setUser(userObject);
+            $state.go('home.matched')
```
Comment (kjrcda) [inline webapp/public/javascripts/controllers.js]: ;

File: webapp/public/javascripts/controllers.js [hunk 8]
```diff
@@ -67,31 +72,44 @@ lunchrControllers.controller('RegisterController', ['$scope', '$http', '$state',
...
-        $http.get('api/users', {params: {email: authService.currentUser()}})
-            .success(function (data, status, headers, config) {
-                $scope.name = (data[0].firstname + " " + data[0].lastname);
-            })
-            .error(function (data, status, headers, config) {
-                ;//couldnt find user in db
-            });
+        var currentUser = authService.currentUser();
+        if(currentUser.matchedWith){
+            $state.go('home.matched')
```
Comment (kjrcda) [inline webapp/public/javascripts/controllers.js]: ;

File: webapp/public/javascripts/controllers.js [hunk 9]
```diff
@@ -67,31 +72,44 @@ lunchrControllers.controller('RegisterController', ['$scope', '$http', '$state',
...
+        }
+        if(currentUser.wantsToBeMatched){
+            $state.go('home.matching')
```
Comment (kjrcda) [inline webapp/public/javascripts/controllers.js]: ;

File: webapp/public/stylesheets/style.css [hunk 10]
```diff
@@ -22,7 +22,29 @@ a {
...
+
+.radiusButtonDiv {
+    float: right;
+}
+
+
+.radiusButton {
+    width: 100px;
+    text-align: left;
+}.div-center-circleBtn {
```
Comment (kjrcda) [inline webapp/public/stylesheets/style.css]: newline? weird how that happened lol

File: webapp/routes/api.js [hunk 11]
```diff
@@ -36,29 +37,31 @@ router.post('/users/register', function (req, res, next) {
         // create and save the user
         var user = new User({
             email: req.body.email, password: req.body.password,
-            firstname: req.body.firstname, lastname: req.body.lastname
-        })
+            firstname: req.body.firstname, lastname: req.body.lastname,
```
Comment (kjrcda) [inline webapp/routes/api.js]: maybe put all fields on separate lines

File: webapp/routes/api.js [hunk 12]
```diff
@@ -36,29 +37,31 @@ router.post('/users/register', function (req, res, next) {
...
+            age: req.body.age, radius: req.body.radius
+        });
 
         user.save(function (error, user) {
             if (error) {
                 return next(error)
```
Comment (kjrcda) [inline webapp/routes/api.js]: ;

File: webapp/routes/socket.js [hunk 13]
```diff
@@ -1,8 +1,68 @@
...
+module.exports = function(socket) {
     socket.on('match', function(data){
-        require('sleep').sleep(2);
-        console.log(data.user);
-        socket.emit('matched' + data.user, {name: 'Brandon'});
+
+        var currentUser = null;
+        User.findOne({email: data.userEmail}, function(error, user) {
+            if(error){
+                return
```
Comment (kjrcda) [inline webapp/routes/socket.js]: ;

File: webapp/routes/socket.js [hunk 14]
```diff
@@ -1,8 +1,68 @@
...
+            currentUser = user;
+            currentUser.wantsToBeMatched = true;
+            currentUser.save(function(err){
+                if(err) {
+                    console.log('Could not save user: ' + currentUser);
+                    console.log(err);
+                    return;
+                }
+                currentUser.password = null;
+                socket.emit('hasBeenMatched', {user: JSON.stringify(currentUser)})
```
Comment (kjrcda) [inline webapp/routes/socket.js:27]: ;

File: webapp/routes/socket.js [hunk 15]
```diff
@@ -1,8 +1,68 @@
...
+            });
+
+        });
+
+        User.find({email: {'$ne': data.userEmail }, wantsToBeMatched: true}, function(error, users){
+            if(error || users.length == 0){
+                //bleh
+                return
```
Comment (kjrcda) [inline webapp/routes/socket.js:35]: ; and remove comment

File: webapp/views/partials/home.jade [hunk 16]
```diff
@@ -5,4 +5,6 @@ div(class='container')
         button(type='button', class='btn-circle', ng-click='match()') Find a Match
 
     div(class='div-center')
-        button(type='button', class='btn btn-block btn-primary', ng-click='editInfo()') Edit my Information
\ No newline at end of file
+        button(type='button', class='btn btn-block btn-primary', ng-click='editInfo()') Edit my Information
+
```
Comment (kjrcda) [inline webapp/views/partials/home.jade:9]: move that comment pertaining to ui-view from other file

**Comment by kyleleung75** [comment]
Everything looked good, just testing it locally.

**Pull request merged**